packages/fetchai/skills/erc1155_deploy/handlers.py

```diff
@@ -0,0 +1,216 @@
+        info_msg = FIPAMessage(
+            message_id=new_message_id,
+            dialogue_reference=("", ""),
+            target=new_target,
+            performative=FIPAMessage.Performative.INFORM,
+            info={
+                "contract": contract.instance.address,
+                "item_ids": contract.item_ids,
+                "trade_nonce": contract_nonce,
+                "from_supply": strategy.from_supply,
+                "to_supply": strategy.to_supply,
+                "value": strategy.value,
+            },
+        )
+
+        self.context.outbox.put_message(
+            to=msg.counterparty,
+            sender=self.context.agent_address,
+            protocol_id=FIPAMessage.protocol_id,
+            message=FIPASerializer().encode(info_msg),
+        )
+
+    def _handle_inform(self, msg: FIPAMessage) -> None:
+        """
+        Handle the INFORM.
+
+        If the INFORM message contains the transaction_digest then verify that it is settled, otherwise do nothing.
+        If the transaction is settled, send the data, otherwise do nothing.
+
+        :param msg: the message
+        :return: None
+        """
+        self.context.logger.info(
+            "[{}]: received INFORM from sender={}".format(
+                self.context.agent_name, msg.counterparty[-5:]
+            )
+        )
+        self.context.logger.info(self.context.ledger_apis.apis.get("ethereum"))
+        strategy = cast(Strategy, self.context.strategy)
+        contract = self.context.contracts.erc1155
+        contract_nonce = cast(int, self.context.shared_state.get("contract_nonce"))
+        tx = contract.get_atomic_swap_single_proposal(
+            from_address=self.context.agent_address,
+            to_address=msg.counterparty,
+            item_id=contract.item_ids[0],
+            from_supply=strategy.from_supply,
+            to_supply=strategy.to_supply,
+            value=strategy.value,
+            trade_nonce=contract_nonce,
+            ledger_api=self.context.ledger_apis.apis.get("ethereum"),
+            skill_callback_id=self.context.skill_id,
+            signature=msg.info.get("signature"),
+        )
+
+        self.context.decision_maker_message_queue.put(tx)
+
+
+class TransactionHandler(Handler):
+    """Implement the transaction handler."""
+
+    SUPPORTED_PROTOCOL = TransactionMessage.protocol_id  # type: Optional[ProtocolId]
+
+    def setup(self) -> None:
+        """Implement the setup for the handler."""
+        pass
+
+    def handle(self, message: Message) -> None:
+        """
+        Implement the reaction to a message.
+
+        :param message: the message
+        :return: None
+        """
+        tx_msg_response = cast(TransactionMessage, message)
+        contract = self.context.contracts.erc1155
+        ledger_api = cast(LedgerApi, self.context.ledger_apis.apis.get("ethereum"))
+        tx_digest = ""
+        if tx_msg_response.tx_id == "contract_deploy":
```

Comment:
See above, these tx_ids could be at least defined in constants, but maybe better in an enum as part of the `contract.py`.